File: pinyougou-goods/src/main/java/com/pinyougou/goods/controller/SpecificationController.java
package com.pinyougou.goods.controller;

import com.github.pagehelper.Page;
import com.pinyougou.common.dto.BaseResponse;
import com.pinyougou.goods.dao.entity.Specification;
import com.pinyougou.goods.dto.SpecificationDTO;
import com.pinyougou.goods.service.ISpecificationService;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/specification")
public class SpecificationController {

	@Autowired
	private ISpecificationService specificationService;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@GetMapping("/findAll")
	@ApiOperation(value="查询全部规格", notes="查询全部规格")
	public BaseResponse findAll(){
		BaseResponse response = new BaseResponse();
		try{
			List<Specification> list = specificationService.findAll();
			response.setData(list);
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@GetMapping("/findPage")
	@ApiOperation(value="分页查询全部规格", notes="分页查询全部规格")
	public BaseResponse findPage(int page,int rows){
		BaseResponse response = new BaseResponse();
		try{
			Page data = specificationService.findPage(page, rows);
			response.setData(data.getResult());
			response.setTotalCount(data.getTotal());
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
	/**
	 * 增加
	 * @param specificationDTO
	 * @return
	 */
	@PostMapping("/add")
	@ApiOperation(value="增加规格", notes="增加规格")
	public BaseResponse add(@RequestBody SpecificationDTO specificationDTO){
		BaseResponse response = new BaseResponse();
		try{
			specificationService.add(specificationDTO);
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
	/**
	 * 修改
	 * @param specificationDTO
	 * @return
	 */
	@PostMapping("/update")
	@ApiOperation(value="修改规格", notes="修改规格")
	public BaseResponse update(@RequestBody SpecificationDTO specificationDTO){
		BaseResponse response = new BaseResponse();
		try{
			specificationService.update(specificationDTO);
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@GetMapping("/findOne")
	@ApiOperation(value="根据id查询规格", notes="根据id查询规格")
	public BaseResponse findOne(Long id){
		BaseResponse response = new BaseResponse();
		try{
			SpecificationDTO dto = specificationService.findOne(id);
			response.setResult(dto);
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@GetMapping("/delete")
	@ApiOperation(value="批量删除规格", notes="批量删除规格")
	public BaseResponse delete(@RequestParam("ids") List<Long> ids){
		BaseResponse response = new BaseResponse();
		try{
			specificationService.delete(ids);
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
		/**
	 * 查询+分页
	 * @param page
	 * @param rows
	 * @return
	 */
	@PostMapping("/search")
	@ApiOperation(value="按条件分页查询规格", notes="按条件分页查询规格")
	public BaseResponse search(@RequestBody Specification specification, int page, int rows  ){
		BaseResponse response = new BaseResponse();
		try{
			Page<Specification> data = (Page<Specification>) specificationService.findPage(specification, page, rows);
			response.setTotalCount(data.getTotal());
			response.setData(data.getResult());
		}catch (Exception e){
			e.printStackTrace();
			response.setErrorMessage(e.getMessage());
		}
		return response;
	}
	
}

File: pinyougou-backend/src/main/resources/static/js/controller/specificationController.js
 //控制层 
app.controller('specificationController' ,function($scope,$controller   ,specificationService){	
	
	$controller('baseController',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		specificationService.findAll().success(
			function(response){
				$scope.list=response.data;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		specificationService.findPage(page,rows).success(
			function(response){
				$scope.list=response.data;	
				$scope.paginationConf.totalItems=response.totalCount;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		specificationService.findOne(id).success(
			function(response){
				$scope.entity= response.result;					
			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var serviceObject;//服务层对象  				
		if($scope.entity.specification.id!=null){//如果有ID
			serviceObject=specificationService.update( $scope.entity ); //修改  
		}else{
			serviceObject=specificationService.add( $scope.entity  );//增加 
		}				
		serviceObject.success(
			function(response){
				if(response.code == "00"){
					//重新查询 
		        	$scope.reloadList();//重新加载
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.batchDelete=function(){
		var ids = $scope.selectedIds;
		console.log(ids);
		if (ids.length == 0) {
			alert("请选择要删除的规格")
			return;
		}
		//获取选中的复选框
		specificationService.delete(ids).success(
			function(response){
				if(response.code == "00"){
					$scope.reloadList();//刷新列表
					$scope.selectedIds=[];
				} else {
					alert(response.message)
				}
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	$scope.search=function(page,rows){
		specificationService.search(page,rows,$scope.searchEntity).success(
			function(response){
				$scope.list=response.data;
				$scope.paginationConf.totalItems=response.totalCount;//更新总记录数
			}			
		);
	}

	// $scope.entity = {specificationOptionList:[]};
	$scope.addTableRow = function() {
		$scope.entity.specificationOptionList.push({})
	}

	$scope.deleteTableRow = function(index) {
		$scope.entity.specificationOptionList.splice(index, 1);
	}
    
});	
